spec/truffle/range/to_a_spec.rb

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 2.0, or
+# GNU General Public License version 2, or
+# GNU Lesser General Public License version 2.1.
+# OTHER DEALINGS IN THE SOFTWARE.
+
+require_relative '../../ruby/spec_helper'
+
+describe "Range#to_a" do
+  it "works for ranges with large endpoints" do
+    large = 1 << 40
+    (large..large+1).to_a.should == [1099511627776, 1099511627777]
+  end
+end
```

Comment:
I'll move this to ruby/spec, it doesn't hurt and it might catch issues on other Ruby impls.
(spec/truffle should only be for code that cannot run other Rubies, e.g., because it needs some Truffle:: method)

Reply:
Do you not think it's a bit unfathomable with the context that some Ruby implementations specialise on ranges? Do you think we should comment or mark it or something?

Reply:
I left a comment about it in the commit message:
```
    Add spec for Range#to_a with 64-bit integers

    * This caught an issue in TruffleRuby, it might be useful for other Rubies too.

diff --git a/spec/ruby/core/range/to_a_spec.rb b/spec/ruby/core/range/to_a_spec.rb
index 08f50e4d9e..9a1352b7b0 100644
--- a/spec/ruby/core/range/to_a_spec.rb
+++ b/spec/ruby/core/range/to_a_spec.rb
@@ -16,6 +16,11 @@ describe "Range#to_a" do
     (0xffff...0xfffd).to_a.should == []
   end

+  it "works with Ranges of 64-bit integers" do
+    large = 1 << 40
+    (large..large+1).to_a.should == [1099511627776, 1099511627777]
+  end
+
   it "works with Ranges of Symbols" do
     (:A..:z).to_a.size.should == 58
   end
```

Indeed, it's a good idea to add a comment either in the spec or in the commit for such cases.